src/Emulator.cpp

```diff
@@ -320,7 +320,16 @@ bool romLoaded(Logger* logger, System system, const std::string& path, void* rom
   
   case System::kAtariLynx:
     rom = util::loadFile(logger, path, &size);
-    RA_OnLoadNewRom((BYTE*)rom + 0x0040, size > 0x0240 ? 0x0200 : size - 0x0040);
+
+    if (!strcmp("LYNX", (char *)rom))
```

Comment:
maybe I'm missing something, but I have a feeling that `!strcmp("LYNX", (char *)rom)` will "always" be true.

Quick test:
```c
#include <stdio.h>
#include <string.h>

int main()
{
    char str1[] = "LYNX",
         str2[] = "LYNX followed by binary data.",
         str3[] = "not starting with LYNX";
    int result;

    // comparing strings str1 and str2
    result = strcmp(str1, str2);
    printf("strcmp(%s, %s) = %d\n", str1, str2, result);

    // comparing strings str1 and str3
    result = strcmp(str1, str3);
    printf("strcmp(%s, %s) = %d\n", str1, str3, result);

    return 0;
}
```
testing:
```
$ ./strcmp
strcmp("LYNX", "LYNX followed by binary data.") = -32
strcmp("LYNX", "not starting with LYNX") = -34
```

I changed my test program above using `strncmp(str1, str2, 4)` and looks like it fives the desired result:
```
$ ./strncmp
strncmp("LYNX", "LYNX followed by binary data.", 4) = 0
strncmp("LYNX", "not starting with LYNX", 4) = -34
```

Reply:
it will work if the fifth byte is 0, but that's a dangerous assumption to make. recommend `memcmp` or individual byte comparisons
```
if (memcmp("LYNX", &rom[0], 4) == 0)
```
```
if (rom[0] == 'L' && rom[1] == 'Y' && rom[2] == 'N' && rom[3] == 'X')
```

Reply:
individual byte comparisons is the [method used in the NES case](https://github.com/RetroAchievements/RALibretro/blob/61963fbc571b9da7f3780986d203571eeda47dbf/src/Emulator.cpp#L236), by the way.

Reply:
`strcmp()` is used in Handy, so I am almost certain that the fifth byte must be 0, anyway.

Reply:
The problem is if the user opens a file that doesn't have any 0 in it. RALibretro will most likely crash in this case.

If the 0 char at the 5th position is part of the signature, I recommend testing it too.

Reply:
I think that `Handy` itself would crash in that case, but OK.